tests/jupyterhub_api/test_check_api.py

```diff
@@ -0,0 +1,79 @@
+"""
+Test if a api is reachable
+"""
+
+import json
+import requests
+import subprocess
+import os
+import docker
+import sys
+sys.path.append("..")
+import time
+from logger import Logger, LOG_FOLDER, LOG_EXTENSION
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
+
+import argparse
+
+def get_args():
+    parser = argparse.ArgumentParser(description='Arguments', formatter_class = argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("--port", dest="port", type=int,
+                        required = True,
+                        )
+    args = parser.parse_args()
+    return args
+
+
+class CheckAPI:
+    def __init__(self, port, verify):
+        self.port = port
+        self.exit =0
+        self.verify = verify
+        self.main_url = "https://localhost:" + str(self.port) + "/hub/api/"
```

Comment:
Jupyterhub can deployed in a different base_path. So this should be configurable (not only in this test).
By base_path I mean: https://localhost:1234/mypathtoswan/hub/api

Reply:
I see, I would add base_path as a parameter.